community/kernel-api/src/main/java/org/neo4j/internal/kernel/api/Read.java

```diff
@@ -265,48 +265,6 @@ long lockingNodeUniqueIndexSeek( IndexReference index, IndexQuery.ExactPredicate
 
     Scan<RelationshipScanCursor> relationshipTypeScan( int type );
 
-    /**
-     * @param nodeReference
-     *         a reference from {@link NodeCursor#nodeReference()}.
-     * @param reference
-     *         a reference from {@link NodeCursor#relationshipGroupReference()}.
-     * @param cursor
-     *         the cursor to use for consuming the results.
-     */
-    void relationshipGroups( long nodeReference, long reference, RelationshipGroupCursor cursor );
```

Comment:
Oh OK, what is the use case for dereferenced access? It complicated implementation and wasn't used so I removed it. My idea of it is that allowing dereferenced access exposes more physical details than passing in a cursor to be placed at that point instead. I can bring it back if you really need it. Would just be fun to learn why it would be preferred over passing in a cursor.

Reply:
I can see that it's now trivial to bring back `relationshipGroups` and `relationships`, but is also `nodeProperties` and `relationshipProperties` necessary? If so why? The thing is that they require additional flags/state to implement which just seems unnecessary to me.